marshal.go

```diff
@@ -91,16 +91,19 @@ func stringSetDecode(b []byte) (map[uint32]*MinimalRepoListEntry, error) {
 	// Length
 	l := r.uvarint()
 	m := make(map[uint32]*MinimalRepoListEntry, l)
+	allBranches := make([]RepositoryBranch, 0, l)
```

Comment:
this can lead to an out of bounds later down the line since l may be smaller than the total number of branches.